presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/UnwrapDateFunctionInPredicate.java

```diff
@@ -0,0 +1,249 @@
+            return rewritten;
+        }
+
+        return tryRewriteFunctionEqualsLiteral(right, left);
+    }
+
+    private Optional<RowExpression> tryRewriteFunctionEqualsLiteral(RowExpression functionSide, RowExpression literalSide)
+    {
+        System.out.println("functionSide: " + functionSide + " (" + functionSide.getClass().getName() + ")");
+        System.out.println("literalSide: " + literalSide + " (" + literalSide.getClass().getName() + ")");
+
+        // Unwrap CAST if present on the literal side
+        if (literalSide instanceof CallExpression) {
+            CallExpression castExpression = (CallExpression) literalSide;
+            if (castExpression.getDisplayName().equalsIgnoreCase("CAST") && castExpression.getArguments().size() == 1) {
+                literalSide = castExpression.getArguments().get(0); // Extract the inner literal
+            }
+        }
+
+        if (!(functionSide instanceof CallExpression) || !(literalSide instanceof ConstantExpression)) {
+            return Optional.empty();
+        }
+
+        CallExpression function = (CallExpression) functionSide;
+        ConstantExpression literal = (ConstantExpression) literalSide;
+
+        String functionName = function.getDisplayName();
+        if (functionName.equalsIgnoreCase(DATE_FUNCTION)) {
+            return rewriteDateFunction(function, literal);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<RowExpression> rewriteDateFunction(CallExpression function, ConstantExpression literal)
+    {
+        System.out.println("Function: " + function + " (" + function.getClass().getName() + ")");
+        System.out.println("Function Arguments: " + function.getArguments());
+        System.out.println("Literal: " + literal + " (" + literal.getClass().getName() + ")");
+        System.out.println("Literal Type: " + literal.getType());
+
+        // Ensure the function has exactly one argument
+        if (function.getArguments().size() != 1) {
+            System.out.println("Function has unexpected number of arguments.");
+            return Optional.empty();
+        }
+
+        RowExpression column = function.getArguments().get(0);
+
+        // Unwrap CAST if present on the column side
+        if (column instanceof CallExpression) {
+            CallExpression castExpression = (CallExpression) column;
+            if (castExpression.getDisplayName().equalsIgnoreCase("CAST") && castExpression.getArguments().size() == 1) {
+                column = castExpression.getArguments().get(0);
+            }
+        }
+
+        // Check that the column is a valid reference
+        if (!(column instanceof VariableReferenceExpression || column instanceof InputReferenceExpression)) {
+            return Optional.empty();
+        }
+
+        // If the literal is of type VARCHAR (io.airlift.slice.Slice), try to parse it as a date string.
+        if (literal.getType().getJavaType().equals(Slice.class)) {
```

Comment:
This is likely a Date-typed constant already. I would just get the value from the `ConstantExpression`